articles/151_node_to_participant_mapping.md

```diff
@@ -0,0 +1,155 @@
+Several approaches can be used to share this information, the proposed approach uses a topic for it.
+Each `Participant` publishes a message with all the information needed to match an entity to a `Node`.
+The message definition is the following:
+
+* ParticipantInfo
+  * gid
+  * NodeInfo
+    * node namespace
+    * node name
+    * reader gid
+    * writed gid
+
+When one entity is updated --e.g.: A `Publisher` is created or destroyed--, a new message is sent.
+
+Identification of `Clients` and `Services` happens according to the ros conventions for their topic names (see [	
+Topic and Service name mapping to DDS](140_topic_and_service_name_mapping.md)).
+
+This topic is considered an implementation detail, and not all `rmw` implementations have to use it.
+Thus, all the necessary logic has to be in the rmw implementation itself or in a downstream package.
+Implementing this logic in `rcl` would make it part of the API, and not an implementation detail.
+
+To avoid code repetition, a common implementation of the logic was done in [rmw_dds_common](https://github.com/ros2/rmw_dds_common/).
+
+#### Details of the ros discovery topic
+
+- topic name: `ros_discovery_info`
+- writer qos:
+  - durability: transient local
+  - history: keep last
+  - history depth: 1
+  - reliability: reliable
+- reader qos:
+  - durability: transient local
+  - history: keep all
+  - history depth: 1
+  - reliability: reliable
+
+## Other implications
+
+#### Security
+
+Previously, each node could have different security artifacts.
+That was possible because each node was mapped to one `Participant`.
+The new approach allows to specify different security artifacts for each process.
+For more details, see [ROS 2 Security Enclaves](ros2_security_enclaves.md).
+
+#### Ignore local publications option
+
+There's an `ignore_local_publications` option that can be set when [creating a subscription](https://github.com/ros2/rmw/blob/2250b3eee645d90f9e9d6c96d71ce3aada9944f3/rmw/include/rmw/rmw.h#L517).
+That option avoids receiving messages from `Publishers` within the same `Node`.
+This wasn't implemented in all the rmw implementations (e.g.: [FastRTPS](https://github.com/ros2/rmw_fastrtps/blob/099f9eed9a0f581447405fbd877c6d3b15f1f26e/rmw_fastrtps_cpp/src/rmw_subscription.cpp#L118)).
+
+After this change, implementing this feature will be less direct.
+Some extra logic needs to be added in order to identify from which `Node` a `Publisher` was created.
+
+## Alternative implementations
+
+### Using keyed topics
+
+Keyed topics could be used, with the participant gid as the key.
+That would allow the reader side of the topic to use a keep last, depth 1 history.
+
+### Using participant/writer/reader user data
+
+Instead of using a custom topic to share the extra ROS specific discovery information, a combination of participant/reader/writer user data could be used.
```

Comment:
@ivanpauno nit: 
```suggestion
Instead of using a custom topic to share the extra ROS specific discovery information, a combination of participant/reader/writer user data could be used:
```
and consider removing the following `e.g.:`.